sktime/forecasting/model_selection/_tune.py

```diff
@@ -46,14 +53,13 @@ def __init__(
         super(BaseGridSearch, self).__init__()
 
     @if_delegate_has_method(delegate=("best_forecaster_", "forecaster"))
-    def update(self, y, X=None, update_params=False):
+    def _update(self, y, X=None, update_params=False):
```

Comment:
Hi @fkiraly and @GuzalBulatova, quick question: How does this work with the decorator, would the object still have `update` from the base class even if the delegate doesn't have an `_update` method? Perhaps these decorators aren't even necessary anymore if all forecasters have these methods or otherwise raise a `NotImplementedError`?

Reply:
I have to look into this again but I think they should still be the public methdods (without the underscore), as we delegate the call to the `best_forecaster_`.

Reply:
hm, but both the best forecaster and the grid search would inherit the same `update`, so it wouldn't really matter?

Reply:
Yes that's what I thought too, but I have a slight preference for delegating the public method, rather than assuming that the forecaster always inherits from `BaseForecaster`

Reply:
yes, good point

Reply:
but the clean way would be for `_update` to delegate to `update`, whic the delegate decorator can't do.
I'd suggest to leave it as is then?

Reply:
Yes, I'd leave it as it is

Reply:
... but then the tests break, apparently? Why? @GuzalBulatova I don't understand this, do you have a good explanation?